<a href="https://colab.research.google.com/github/flamorim/BI-Master-2021.1/blob/Features(versao-1)/Iris_solucao.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Análise de Previsão de Churn


Neste script iremos prever a espécie de uma planta usando o algoritmo KNN (K Nearest Neighbor).

# Setup

##Importar bibliotecas necessárias

In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score

# Exercício 1

Carregue o arquivo csv chamado 'Iris.csv' usando pandas e salve a base em uma variável chamada df. (dica: pd.read_csv('<nome_da_base.csv>')) Imprima as primeiras 10 linhas do dataframe e verifique se tudo foi carregado corretamente. (dica: df.head())

In [11]:
df = pd.read_csv('Iris.csv')

In [5]:
df.head()

,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
0,5.1,3.5,1.4,0.2,Iris-setosa
1,4.9,3.0,1.4,0.2,Iris-setosa
2,4.7,3.2,1.3,0.2,Iris-setosa
3,4.6,3.1,1.5,0.2,Iris-setosa
4,5.0,3.6,1.4,0.2,Iris-setosa


# Exercício 2: 

Que coluna indica a espécie da planta? Quantas plantas existem na sua base de dados? (dica: len(?))


In [6]:
#coluna rótulo: 'Species'
# número de plantas:
len(df)

150

# Exercício 3:

Use df.describe() para explorar cada coluna. Todos devem ser igual a 150 pois não temos missing values!!

In [7]:
df.describe()

,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.054000,3.758667,1.198667
std,0.828066,0.433594,1.764420,0.763161
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


# Exercício 5:

Verifique o número de colunas da base de dados (dica: df.shape)

Separe a base de dados em atributos de entrada e rótulos. Associe os atributos a variável X e os rótulos a variável y.

Verifique as primeiras linhas das bases separadas.

In [8]:
df.shape

(150, 5)

In [ ]:
X = df.iloc[:,0:4]
y = df.iloc[:,4]

In [ ]:
X.head()

In [ ]:
y.head()

# Exercício 6: 

Separa a base de treino (70%) e teste (30%) e associe-as as variáveis X_train, X_test, y_train, y_test (dica: train_test_split(X,y, test_size=0.3))

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.3)

# Exercício 7: 

Crie um classificador KNN com 5 vizinhos. (hint: KNeighborsClassifier(n_neighbors=5))

In [ ]:
knn = KNeighborsClassifier(n_neighbors=5)

# Exercício 8:

Apresente a base de treino para ajustar o modelo. (hint: knn.fit(<'base_de_treino_atributos'>, <'base_de_treino_rotulos'>))

In [ ]:
knn = knn.fit(X_train, y_train)

# Exercício 9:

Use o modelo para fazer previsões na base de teste. Associe os resultados de previsão a variável y_pred

In [ ]:
y_pred = knn.predict(X_test)

# Exercício 10:

Determine quão acurado é o seu modelo. (dica: accuracy_score(<'rotulo_real'>, <'rotulo_previsto'>))

In [ ]:
 accuracy_score(y_test,y_pred)